tests/test_device.py

```diff
@@ -246,7 +245,7 @@ def mock_submit_job(*args):
             pass
```

Comment:
In `TestDevice` above, please add a test explicitly checking the initialization when specifying a `backend` kwarg, as well as the default behaviour when “backend=None”